dll/ntdll/ldr/ldrapi.c

```diff
@@ -1014,7 +1014,7 @@ LdrQueryProcessModuleInformationEx(IN ULONG ProcessId,
         ModuleListHead = &NtCurrentPeb()->Ldr->InLoadOrderModuleList;
         Entry = ModuleListHead->Flink;
 
-        while (Entry != ModuleListHead)
+        while (Entry != ModuleListHead && Status == STATUS_SUCCESS)
```

Comment:
With this change we no longer return the required size if the buffer is too small, instead we abort immediately. I think Windows does something similar (also fails early), but it would be nice to have a test to prove it.

Reply:
The idea here was to not use `ModulePtr` uninitialized, I'll solve that another way then :)